app/components/inputs/InputComponent.js

```diff
@@ -0,0 +1,194 @@
+      showErrors,
+      invalidMessage,
+      requiredMessage,
+      required,
+      invalid,
+      value,
+      placeholder,
+      onChange,
+      disabled,
+      readOnly,
+      unit,
+      hidden,
+      type,
+    } = this.props;
+    return (
+      hidden ? null :
+        <Aux>
+          <input id="parsedInput"
+            style={{ display: "none" }}
+            ref={input => { this.input = input; }}
+            type="number"
+            className="input"
+            disabled={disabled ? disabled : null}
+            readOnly={readOnly ? readOnly : null}
+            placeholder={placeholder}
+            value={isNullOrUndefined(value) ? "" : value}
+            onChange={onChange}
+            onFocus={this.onInputFocus}
+            onBlur={this.onInputBlur}
+            onKeyDown={this.onKeyDown}
+          />
+          <div className={showErrors && (invalid && value || required && !value) ? this.state.divClassName + " error" : this.state.divClassName} ref={div => { this.state.inputUnitDiv = div; }}>
+            <div contentEditable="true"
+              ref={input => { this.input = input; }}
+              type={type || "text"}
+              className="input"
+              disabled={disabled ? disabled : null}
+              readOnly={readOnly ? readOnly : null}
+              placeholder={placeholder}
+              value={isNullOrUndefined(value) ? "" : value}
+              onChange={this.sendToParser}
+              onMouseOver={this.onChange}
+              onBlur={this.onChange2}
+              onKeyDown={this.onKeyDown2}
+            >
+              <span id="parsedInputDisplay" style={{ display: "none", marginTop: "2px;" }}
+                onKeyDown={this.onKeyDown2}
+              ></span>
+              <div id="viewInputDisplay">
+                <span
+
+                  type="number" id="input-main" className="input-main">0.00</span>
+                <span type="number" id="input-sub" className="input-sub">0000000</span>
+              </div>
+            </div>
+            {unit && !(showErrors && ((invalid && value) || (required && !value))) ? <div className="unit-area">{unit}</div> : null}
+          </div>
+          {showErrors ? (
+            <div className="input-errors-area">
+              {invalid && value ? (
+                <div className="input-error">
+                  {invalidMessage ? invalidMessage :
+                    <T id="input.invalidInput" m="This field is wrong" />}
+                </div>
+              ) : null}
+              {required && !value ? (
+                <div className="input-error">
+                  {requiredMessage ? requiredMessage :
+                    <T id="input.requiredInput" m="This field is required" />}
+                </div>
+              ) : null}
+            </div>
+          ) : null}
+        </Aux>
+    );
+  }
+}
+
+function pad(str, max) {
+  str = str.toString();
+  return str.length < max ? pad("0" + str, max) : str;
```

Comment:
Using recursion for a simple pad function seems like overkill to me. You can also just use `str.padStart(max, "0")` instead of having to write this pad function.

Reply:
Updated